### DISCUSSION TITLE
Shape radius different with RoundedCornersTransformation size

### DISCUSSION THREAD

**Discussion by anthony3637** [coil-kt/coil#1289] [General]
**Describe the bug**
1: we used the shape file  to view's background
2: we used coil's RoundedCornersTransformation 

but radius size difference, how to resolve it

shape.xml
`<?xml version="1.0" encoding="utf-8"?>
<shape xmlns:android="http://schemas.android.com/apk/res/android"
    android:shape="rectangle">

    <solid android:color="#22229900"/>
    <corners android:radius="50dp"/>

</shape>`

activity xml:
 `![image](https://user-images.githubusercontent.com/4946107/170238402-3aa75c0e-3acd-49bb-ad7d-e9aaaee2324e.png)
`

activity code:
`val current = TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, 50f, resources.displayMetrics)
 binding.img.load("xxx.jpg"){
            transformations(RoundedCornersTransformation(current))
        }
`

finally, we found  different radius

**Comment by marazmone** [comment]
@anthony3637 did you find the solution?

**Comment by guuilp** [comment]
I'm moving from Glide to Coil and I saw this strange behavior too. In my case, I'm using 12f of Radius and the result is different between the libraries.

Since I saw some division by two on the RoundedCornersTransformation code, I tried to multiply my 12f radius by 2 and voilà: same result as Glide.

So the RoundedCornersTransformation is doing some wrong math.

**Comment by colinrtwhite** [comment]
Hmm looks like Glide uses `canvas.drawRoundRect` whereas Coil uses `drawPath` with a rounded rect. I'll take a look at this, though I'm worried changing the radius calculation now could break the way it works for existing users.

**Reply by guuilp** [reply]
Yeah, that won't be a easy decision to make.

Here the team uses the radius written by the designer on Figma. Glide visually reproduces the same rounding as the design, but Coil doesn't.

Maybe introduce a parameter on the transformation constructor to decide how to round the corners?

For me is not a big issue since we wrap coil in our on image-loader library: I added a multiply by 2 in there and it worked just as Glide used to work for us.
